### ISSUE TITLE
Raft: new leader should clear synchro queue  **strictly** before becoming writeable

### ISSUE THREAD

**Issue by sergepetrenko** [tarantool/tarantool#5440]
Found a test failure:
https://gitlab.com/tarantool/tarantool/-/jobs/804537914#L4837
```
4835[054] replication/election_qsync.test.lua             vinyl           [ fail ]
4836[054] 
4837[054] Test failed! Result content mismatch:
4838[054] --- replication/election_qsync.result	Sat Oct 17 16:25:22 2020
4839[054] +++ replication/election_qsync.reject	Wed Oct 21 17:04:52 2020
4840[054] @@ -127,14 +127,15 @@
4841[054]   | ...
4842[054]  _ = box.space.test:replace{2}
4843[054]   | ---
4844[054] + | - error: Found uncommitted sync transactions from other instance with id 2
4845[054]   | ...
4846[054]  box.space.test:select{}
4847[054]   | ---
4848[054]   | - - [1]
4849[054] - |   - [2]
4850[054]   | ...
4851[054]  box.space.test:drop()
4852[054]   | ---
4853[054] + | - error: Found uncommitted sync transactions from other instance with id 2
4854[054]   | ...
4855[054]  
```

This happens because the instance first becomes a leader, and only then fires the on update triggers, which are trying to clear limbo from an old leader. So it's possible for a user to wait till the instance is leader and still fail inserting some synchronous data because the instance hasn't cleared limbo yet.

Raw idea: need to rework raft to set its state and fire on_update triggers in the same fiber without yields.

**Comment by Totktonada** [comment]
@Gerold103 @avtikhon Does it fix #5430 and #5395?

**Comment by Gerold103** [comment]
Hm. Probably. Thanks for noticing! @sergepetrenko, could you please, make a follow-up patch to unmark these tests as flaky?

**Comment by avtikhon** [comment]
> @Gerold103 @avtikhon Does it fix #5430 and #5395?

Checked that it didn't fix #5430
